pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/model/XPathSuggestions.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.util.fxdesigner.model;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.List;
+
+
+
+
+public class XPathSuggestions {
+    private List<String> xPathSuggestions = new ArrayList<>();
+    private String language;
+    private List<String> listOfSuggestions;
+
+
+    public XPathSuggestions(String language) {
+        this.language = language.replaceAll("[0-9]", "").replaceAll("//s", "").trim().toLowerCase();
+        try {
+            this.xPathSuggestions = createList(getClasses("net.sourceforge.pmd.lang."
+                                                               + language.replaceAll("[0-9]", "").replaceAll("//s", "").trim().toLowerCase() + ".ast"));
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    public List<String> getXPathSuggestions(String input) {
+        List<String> resultsToDisplay = new ArrayList<>();
+        for (String s: xPathSuggestions) {
+            if (s.contains(input)) {
+                resultsToDisplay.add(s);
+            }
+        }
+        return resultsToDisplay;
+    }
+
+    private List<String> createList(Class[] classArray) {
+        List<Class> fileNameList = Arrays.asList(classArray);
+
+        for (Class c : fileNameList) {
+            if (c.getSimpleName().startsWith("AST")) {
+                xPathSuggestions.add(c.getSimpleName().substring("AST".length()));
```

Comment:
Don't add directly to the field, this method should build a new list and return it without touching the fields. You can use streams to do that in four lines.

Reply:
This is still pending